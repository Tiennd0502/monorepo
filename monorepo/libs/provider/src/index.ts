export { default as TamaguiProvider } from './TamaguiProvider';
export { default as QueryClientProvider } from './QueryClientProvider';
