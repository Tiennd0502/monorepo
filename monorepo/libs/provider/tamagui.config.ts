import { createTamagui, createTokens } from 'tamagui';
import { config } from '@tamagui/config/v3';
import { themes } from '@tamagui/themes';

import {
  colorsMapping,
  radius,
  size,
  space,
  zIndex,
  defaultFont,
  headingFont,
} from '@monorepo/ui';

const tokens = {
  radius,
  size,
  space,
  zIndex,
};

const tamaguiConfig = createTamagui({
  ...config,
  defaultFont: 'body',
  themes,
  fonts: {
    body: defaultFont,
    heading: headingFont,
  },
  tokens: createTokens({
    ...tokens,
    color: colorsMapping,
  }),
});

type AppConfig = typeof config;

declare module 'tamagui' {
  type TamaguiCustomConfig = AppConfig;
}

export default tamaguiConfig;
