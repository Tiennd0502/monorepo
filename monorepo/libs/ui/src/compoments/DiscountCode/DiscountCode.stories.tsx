import { Meta, StoryObj } from '@storybook/react';
import { Stack } from 'tamagui';

import DiscountCode from '.';

const meta: Meta<typeof DiscountCode> = {
  title: 'Components/DiscountCode',
  component: DiscountCode,
  decorators: [
    (Story) => (
      <Stack width={400}>
        <Story />
      </Stack>
    ),
  ],
};

export default meta;

type Story = StoryObj<typeof DiscountCode>;

export const Default: Story = {
  args: {
    value: '',
    onSubmit: (value: string) => null,
  },
};

export const Code: Story = {
  args: {
    value: '#23fw12',
    onSubmit: (value: string) => null,
  },
};
