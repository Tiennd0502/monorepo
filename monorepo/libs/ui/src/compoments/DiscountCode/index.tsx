import { memo, useCallback, useState } from 'react';
import { Stack } from 'tamagui';

// Themes | Components
import Input from '../Input';
import IconButton from '../IconButton';

import { shadows } from '../../themes';
import { ChevronRightIcon, DeleteIcon } from '../icons';

interface DiscountCodeProps {
  value?: string;
  onSubmit: (value: string) => void;
}

const DiscountCode = ({ value = '', onSubmit }: DiscountCodeProps) => {
  const [code, setCode] = useState<string>(value);
  const [isDeleteIcon, toggleIcon] = useState<boolean>(!!value);

  const handleSubmit = useCallback(() => {
    onSubmit(code);
    toggleIcon((prev) => !prev);
  }, [code, onSubmit]);

  const handleChangeText = useCallback((text: string) => {
    setCode(text);
  }, []);

  const handleResetValue = useCallback(() => {
    if (code) {
      setCode('');
      onSubmit('');
      toggleIcon((prev) => !prev);
    }
  }, [onSubmit, code]);

  return (
    <Stack position="relative" zIndex="$2">
      <Input
        testID='input'
        position="absolute"
        top={0}
        width="100%"
        height="$11"
        value={code}
        variant="flushed"
        paddingVertical={10}
        backgroundColor="$secondary"
        placeholder="Enter your promo code"
        borderRadius="$2"
        style={shadows.input}
        onChangeText={handleChangeText}
      />
      <IconButton
        testID='icon'
        solid
        position="absolute"
        top={0}
        zIndex={2}
        right={0}
        height="$11"
        width="$11"
        borderRadius="$2"
        backgroundColor="$primary"
        onPress={isDeleteIcon ? handleResetValue : handleSubmit}
      >
        {isDeleteIcon ? (
          <DeleteIcon color="$secondary" />
        ) : (
          <ChevronRightIcon color="$secondary" />
        )}
      </IconButton>
    </Stack>
  );
};

export default memo(DiscountCode);
