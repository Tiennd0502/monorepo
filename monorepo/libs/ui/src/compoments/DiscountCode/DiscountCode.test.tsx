import {fireEvent, render, waitFor} from '@testing-library/react-native';

import { TamaguiProvider } from '@monorepo/provider';
import DiscountCode from '.';

const onSubmit = jest.fn();
const baseProps = {onSubmit};

const setup = (moreProps = {}) => {
  const props = {
    ...baseProps,
    ...moreProps,
  };
  
  return render(
    <TamaguiProvider>
      <DiscountCode {...props} />
    </TamaguiProvider>
  );
};

describe('DiscountCode', () => {
  const onSubmit = jest.fn();
  const promoCode = '#123456';

  it('should render DiscountCode component', () => {
    const container = setup();

    expect(container).toMatchSnapshot();
  });

  it('should render DiscountCode component has value', () => {
    const container = setup({value: promoCode});

    expect(container).toMatchSnapshot();
  });

  it('calls onSubmit after entering input and clicked chevron right icon', () => {
    const {getByTestId} = setup();
    waitFor(() => {
      const input = getByTestId('input');
      const ChevronRightIcon = getByTestId('icon');
  
      fireEvent.changeText(input, promoCode);
      fireEvent.press(ChevronRightIcon);
  
      expect(onSubmit).toHaveBeenCalled();
    })
  });

  it('calls onSubmit when delete icon clicked', () => {
    const {getByTestId} = setup({value: promoCode});
    waitFor(() => {
      const deleteIcon = getByTestId('icon');
      fireEvent.press(deleteIcon);

      expect(onSubmit).toHaveBeenCalled();
    })
  });
});
