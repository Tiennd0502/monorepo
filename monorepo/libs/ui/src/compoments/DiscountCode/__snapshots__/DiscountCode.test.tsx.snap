// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiscountCode should render DiscountCode component 1`] = `
<View
  style={
    {
      "position": "relative",
      "zIndex": 200,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "position": "relative",
        "zIndex": 1,
      }
    }
  >
    <TextInput
      allowFontScaling={false}
      focusVisibleStyle={{}}
      focusable={true}
      onChangeText={[Function]}
      placeholder="Enter your promo code"
      placeholderTextColor="#999"
      style={
        {
          "backgroundColor": "#fff",
          "borderBottomColor": "transparent",
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderBottomWidth": 1,
          "borderLeftColor": "transparent",
          "borderLeftWidth": 1,
          "borderRightColor": "transparent",
          "borderRightWidth": 1,
          "borderStyle": "solid",
          "borderTopColor": "transparent",
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "borderTopWidth": 1,
          "color": "textPrimary",
          "elevation": 50,
          "flex": 1,
          "fontFamily": "Tahoma, Helvetica, Arial, sans-serif",
          "fontSize": 16,
          "height": 44,
          "minWidth": 0,
          "paddingBottom": 10,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 10,
          "position": "absolute",
          "shadowColor": "$backgroundTertiary",
          "shadowOffset": {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 1,
          "top": 0,
          "width": "100%",
        }
      }
      testID="input"
      value=""
    />
  </View>
  <View
    accessibilityRole="button"
    focusVisibleStyle={{}}
    focusable={true}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#242424",
        "borderBottomColor": "transparent",
        "borderBottomLeftRadius": 8,
        "borderBottomRightRadius": 8,
        "borderBottomWidth": 0,
        "borderLeftColor": "transparent",
        "borderLeftWidth": 0,
        "borderRightColor": "transparent",
        "borderRightWidth": 0,
        "borderStyle": "solid",
        "borderTopColor": "transparent",
        "borderTopLeftRadius": 8,
        "borderTopRightRadius": 8,
        "borderTopWidth": 0,
        "flexDirection": "row",
        "flexWrap": "nowrap",
        "height": 44,
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
        "position": "absolute",
        "right": 0,
        "shadowColor": "rgb(0,0,0)",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
        "top": 0,
        "width": 44,
        "zIndex": 2,
      }
    }
    testID="icon"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="24"
      bbWidth="24"
      fill="none"
      focusable={false}
      height="24"
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      vbHeight={24}
      vbWidth={24}
      width="24"
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M9 18l6-6-6-6"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          propList={
            [
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4294967295,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth="1.5"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`DiscountCode should render DiscountCode component has value 1`] = `
<View
  style={
    {
      "position": "relative",
      "zIndex": 200,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "position": "relative",
        "zIndex": 1,
      }
    }
  >
    <TextInput
      allowFontScaling={false}
      focusVisibleStyle={{}}
      focusable={true}
      onChangeText={[Function]}
      placeholder="Enter your promo code"
      placeholderTextColor="#999"
      style={
        {
          "backgroundColor": "#fff",
          "borderBottomColor": "transparent",
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderBottomWidth": 1,
          "borderLeftColor": "transparent",
          "borderLeftWidth": 1,
          "borderRightColor": "transparent",
          "borderRightWidth": 1,
          "borderStyle": "solid",
          "borderTopColor": "transparent",
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "borderTopWidth": 1,
          "color": "textPrimary",
          "elevation": 50,
          "flex": 1,
          "fontFamily": "Tahoma, Helvetica, Arial, sans-serif",
          "fontSize": 16,
          "height": 44,
          "minWidth": 0,
          "paddingBottom": 10,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 10,
          "position": "absolute",
          "shadowColor": "$backgroundTertiary",
          "shadowOffset": {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 1,
          "top": 0,
          "width": "100%",
        }
      }
      testID="input"
      value="#123456"
    />
  </View>
  <View
    accessibilityRole="button"
    focusVisibleStyle={{}}
    focusable={true}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#242424",
        "borderBottomColor": "transparent",
        "borderBottomLeftRadius": 8,
        "borderBottomRightRadius": 8,
        "borderBottomWidth": 0,
        "borderLeftColor": "transparent",
        "borderLeftWidth": 0,
        "borderRightColor": "transparent",
        "borderRightWidth": 0,
        "borderStyle": "solid",
        "borderTopColor": "transparent",
        "borderTopLeftRadius": 8,
        "borderTopRightRadius": 8,
        "borderTopWidth": 0,
        "flexDirection": "row",
        "flexWrap": "nowrap",
        "height": 44,
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
        "position": "absolute",
        "right": 0,
        "shadowColor": "rgb(0,0,0)",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
        "top": 0,
        "width": 44,
        "zIndex": 2,
      }
    }
    testID="icon"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="24"
      bbWidth="24"
      fill="none"
      focusable={false}
      height="24"
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      vbHeight={24}
      vbWidth={24}
      width="24"
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M3 6h18M8 6V4a2 2 0 012-2h4a2 2 0 012 2v2m3 0v14a2 2 0 01-2 2H7a2 2 0 01-2-2V6h14z"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          propList={
            [
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4294967295,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth="2"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;
